simulate/simulate.cc

```diff
@@ -2632,18 +2632,42 @@ void Simulate::Render() {
   }
 
   // user figures
+  if (this->newfigurerequest.load() == 1) {
+    this->user_figures_.clear();
+    for (const auto& [viewport, figure] : this->user_figures_new_) {
```

Comment:
Can this loop be replaced with user_figures_ = user_figures_new_?

Or better:
```
user_figures_.clear();
std::swap(user_figures_, user_figures_new_);
```

Which performs no copies (just swaps the underlying pointers).

Same for the other loops.

Reply:
Done